master/buildbot/process/buildstep.py

```diff
@@ -797,19 +798,18 @@ def failed(self, why):
         if why.check(BuildStepFailed):
             self.finished(FAILURE)
             return
-
         klog.err_json(why, "BuildStep.failed; traceback follows")
         results = EXCEPTION
         try:
             if self.progress:
                 self.progress.finish()
             try:
+                if isinstance(why.type, str):
```

Comment:
I think it is a little bit ugly to fix it in this way. So basically right now, we just try to cure a symptom, not a root cause of sickness. The symptom is that the error does not have type field (or has the wrong type), so we inject this attribute. But the whole problem is why this object does have wrong type field. And the root of the problem is in a separate place.

Right now is ok, we just incur technical debt, but we should create a refactor task for it on Jira :)

Reply:
https://jira.hq.unity3d.com/browse/CDSK-984